motor/input/input.h

```diff
@@ -0,0 +1,52 @@
+#ifndef _MOTOR_INPUT_INPUT_H_
+#define _MOTOR_INPUT_INPUT_H_
+
+#include <vector>
+
+#include "motor/event.h"
+
+namespace motor::input
+{
+// TODO(kadircet): the following should come from enums, once we have them...
+constexpr const size_t kNumKeys = 256;
+constexpr const size_t kNumAxes = 8;
+
+enum DeviceType
+{
+  UNKNOWN_DEVICE = 0,
+  // TODO(kadircet): Handle multiple devices of same type.
+  KEYBOARD_0,
+  MOUSE_0,
+  GAMEPAD_0,
+};
+
+struct KeyInput
+{
+  DeviceType device_id_ = UNKNOWN_DEVICE;
+  // TODO(kadircet): Come up with an enum for these.
+  int id_ = 0;
+  bool is_pressed_ = false;
```

Comment:
We may use the term "down" instead of "pressed", since usually press means pressing and releasing something in UI terms.

Reply:
Done